web/src/main/java/com/epam/rft/atsy/web/exceptionhandling/UncheckedExceptionResolver.java

```diff
@@ -35,7 +51,19 @@ public ModelAndView resolveException(HttpServletRequest httpServletRequest,
       return PASS_TO_NEXT_RESOLVER;
     }
 
-    httpServletResponse.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+    int httpStatusCode =
+        AnnotationUtils.isAnnotationDeclaredLocally(ResponseStatus.class, e.getClass())
+            ? AnnotationUtils.findAnnotation(e.getClass(), ResponseStatus.class).value().value()
+            : HttpServletResponse.SC_INTERNAL_SERVER_ERROR;
+
+    String
+        errorMessageKey =
+        errorMessageKeyMap.getOrDefault(e.getClass(), TECHNICAL_ERROR_MESSAGE_KEY);
+    String errorMessage = messageKeyResolver.resolveMessageOrDefault(errorMessageKey);
+
+    httpServletResponse.setStatus(httpStatusCode);
+    httpServletRequest.setAttribute(WebUtils.ERROR_STATUS_CODE_ATTRIBUTE, httpStatusCode);
```

Comment:
Please use `modelAndView.addObject(requestAttributeName, value)` construct.

This needs some modification of the current execution path, like move the creation of the ModelAndView instance above the httpStatusCode retrieval.